### PR TITLE
Remove fastapi-utls

### PR DESCRIPTION
**Pull request by jdavcs**
Ref #17185, #12541 

This removes the dependency on fastapi-utils that is a blocker for SQLAlchemy 2.0. See #17185 for rationale. There has also been a discussion on the backend channel on Dec 15 ([ref](https://app.element.io/#/room/#galaxyproject_backend:gitter.im/$GEB3pKCl4HTmX4jFQxXHpSgrbPuf9bwKcnsywS7Ple0))

To simplify the review of this PR (re: discussion on the channel), I've split it into the following commits:
- One commit per module to remove the class decorator, move the class attribute(s) to the methods, and remove the `self` parameter from the method definition (thus making the methods static). There is almost no indentation change here, which, I hope, will simplify reviewing.
- One final commit that reformats the updated modules by ONLY removing the class definition, thereby converting the static methods into standalone functions. (Indentation is changed here, so everything is marked as red)

A few exceptions:
- In a few cases there were helper methods (prefixed with an underscore) that were called from the endpoint-handling methods. I moved those our of the class and passed the resolved dependencies to them directly (as it should have been done in the first place, and indeed *was* already done in some cases)
- There are a few final commits that remove the `cbv` definition and update the dependencies.
- `item_tags` still has a class. However, here we have 3 classes dynamically generated, and keeping the endpoint-handling methods as static methods works just as fine.

So, overall, the process went as follows:
1. Pick a module that uses `cbv`
2. Remove the decorator
3. Move the dependency defined on the class to each endpoint-handling method; remove the `self` parameter from each method.
5. Change the call to the dependency inside the method (remove `self.`)
6. If there were any helper methods, move them out of the class, remove the `self` parameter, and pass the dependency to them. 
7. Repeat while `grep cbv` is not empty.
8. Replace classes with standalone functions
9. Update dependencies.

I hope this helps.

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

File: lib/galaxy/dependencies/dev-requirements.txt [hunk 1]
```diff
@@ -2,16 +2,16 @@ aiohttp==3.9.1 ; python_version >= "3.8" and python_version < "3.12"
 aiosignal==1.3.1 ; python_version >= "3.8" and python_version < "3.12"
 alabaster==0.7.13 ; python_version >= "3.8" and python_version < "3.12"
 amqp==5.2.0 ; python_version >= "3.8" and python_version < "3.12"
-anyio==4.1.0 ; python_version >= "3.8" and python_version < "3.12"
```
Comment (nsoranzo) [inline lib/galaxy/dependencies/dev-requirements.txt]: I'd only remove fastapi-utils without all the unrelated dependencies updates.

Comment (jdavcs) [reply lib/galaxy/dependencies/dev-requirements.txt]: Absolutely! I think I ran the update script, but I don't have to. I'll fix this.

Comment (jdavcs) [reply lib/galaxy/dependencies/dev-requirements.txt]: Done.

**Review by mvdbeek — COMMENTED** [review]
I don't think this is heading the right direction, as discussed extensively on the backend channel.

**Comment by jdavcs** [comment]
> I don't think this is heading the right direction, as discussed extensively on the backend channel.

I've provided a summary of my point in the [accompanying issue](https://github.com/galaxyproject/galaxy/issues/17185#issuecomment-1860863878).

**Comment by jdavcs** [comment]
Superseded by #17205

**Pull request closed without merging**